src/main/kotlin/stringcalculator/domain/Calculator.kt

```diff
@@ -0,0 +1,27 @@
+package stringcalculator.domain
+
+import java.util.function.BinaryOperator
+
+enum class Calculator(private val sign: String, private val operation: BinaryOperator<Int>) {
+
+    ADD("+", { a, b -> a + b }),
+    SUBTRACT("-", { a, b -> a - b }),
+    MULTIPLY("*", { a, b -> a * b }),
+    DIVIDE("/", { a, b -> a / b });
+
+    companion object {
+        fun of(sign: String): Calculator {
+            values().find { it.sign == sign }
+                ?.let { return it }
+                .run { throw IllegalArgumentException("연산할 수 없는 기호입니다. 기호: $sign") }
```

Comment:
`scope function`을 활용한 구현 너무 좋았어요. 👍👍

하지만 지금의 케이스는 scope function 사용을 하지 않고 간단하게 아래와 같이 변경할 수 있을 것 같아요. 😃

```suggestion
            values().find { it.sign == sign } ?: throw IllegalArgumentException("연산할 수 없는 기호입니다. 기호: $sign")
```


[apply, run, with, let, also 차이 한 번에 정리하기](https://kotlinworld.com/255)

Reply:
자바의 orElse 메서드와 비슷한걸 찾지 못했었는데 `?:` Elvis operator 라는 연산자가 있네요!